### ISSUE TITLE
Load and Save still broken after changing Compatibility settings

### ISSUE THREAD

**Issue by juanitogan** [juanitogan/rbxit#9]
@TheRaccoonKing -- Good news and bad news for you.  The good news is that I was able to break one of my test installs like how you are experiencing.  That bad news for you is that I was able to recover from it exactly according to my theory in the troubleshooting section.  I've tweaked the wordage a bit and included here for your convenience in case you want to give it another go.

1. Run `CSTORM-Compatibility-Unset.bat`
2. Unset everything in the compatibility tab for CSTORM.EXE
3. Reboot (appears to be necessary for some reason I can't explain)
4. Make sure the compatibility tab settings are still unset (don't run the exe yet), otherwise, goto step 2
5. Run `CSTORM-Compatibility-Set.bat`
6. Run `CSTORM.EXE` and good luck!

If that doesn't work, see the additional notes I added after that in the troubleshooting section about other install folders and reading the registry.

**Comment by TheRaccoonKing** [comment]
@juanitogan I really appreciate all the effort you've put into this.

Unfortunately, nothing seems to be working for me.  I checked each of those registry keys, and found paths for several other games, but nothing for Cyberstorm.

I did try other paths, but only after I was using your system, and honestly I can't remember where I put the install in those cases.  And since the registry has no info, I doubt I could recreate them anyway.

Thanks again for all your help.  If you make any future discoveries, please let me know.  Otherwise, I'll survive without it!

I'm sure many other fans appreciate all your hard work on this.

**Comment by juanitogan** [comment]
Yeah, sorry about that.  I was just now figuring out myself that I spoke too soon.  It turns out the behavior I was recently experiencing was due to testing from Explorer++ instead of File Explorer.  In File Explorer (or any other Windows way of launching a program) CSTORM wasn't broken at all and so I can't say that method above works after all.  So, I've had to reverse some of the language in the wiki yet again (what I was doing when your message came in).

Still, it helps that I now know that Explorer++ and (likely) similar programs can mess with all this.  You helped me get that far.

Anyhow, thank you for your willingness to test all this.  Even though it hasn't helped you yet, your testing has helped me slowly narrow this issue down.  It is particularly helpful to know that you didn't find anything in those registry keys.  This should help me focus on things other than path-conflicts and/or compat setting conflicts.

Are you willing to run other tests if I come up with some?  In particular, I'm thinking about writing a batch script that will output anything in your registry (a read-only operation) related to Cyberstorm so I can look at it.  Or, if running such a batch file makes you uneasy, there are some solid free tools for doing bulk searches yourself such as RegScan and RegistryFinder.

**Comment by TheRaccoonKing** [comment]
Yeah, I appreciate the help, and I'm happy to do what I can.

What are we looking for, and how do I find it?

**Comment by juanitogan** [comment]
Sorry for the delay.  Been busy working with the disassembled machine code looking for a better solution.  I've worked around one error only to work around another and run into a third.  I can now get through the load dialog before crashing.  No telling where it will stop and I'm not finding spare bytes in all places to keep adding workarounds at the machine code level.

So, anyhow, back to this issue at hand.  I assume your Win10 is 64-bit?  Regardless, what we are basically looking for is why your system is not responding to the compat settings in the SDB file supplied with the patch.  It could be an interfering program like Explorer++, or something else low level like antivirus programs (I use NOD32), or backup programs (I use CrashPlan for now).  It could be something with your user privs or folder privs (I assume you are logged in as an admin user and not a standard user).  It could be a problem with the SDB install or some other compat trouble.

I can't check all of that from here but the following will let me have a good look at some of it, if you are willing:

Download RegScanner here:
http://www.nirsoft.net/utils/regscanner.html

You don't need to get the installer if you don't want to "install" it.  You can just unzip one of the zips and run it (the sign of a good tool, if you ask me).

Here are the settings I need:

- Find String: CSTORM CyberStorm e5bcabf7 .CBM .cbs .hrc
- Matching: Registry item contains one or more words.
- Uncheck: [_] Data   (occasionally useful but also generates way too much information)
- Select all keys in: [x] Scan the following base keys

![regscanner-options](https://user-images.githubusercontent.com/2923222/33402438-cf06ed82-d519-11e7-8bab-13383062bdf7.png)
(How did that get so orange?  Weird.)

After the search runs, select all the results (Ctrl+A), then use the Save Selected Items menu (Ctrl+S) to save as a CSV file (found under Save As Type in the file dialog).  Then attach the file, or paste the contents of the file, here.

**Comment by juanitogan** [comment]
So, good news, never mind on all this.  That third breakage point turned out to be the last one once I got through it yesterday.  So I'll be releasing a new patch soon once I finish testing it and package it.  It will eliminate the need for the most problematic compat settings -- and DxWnd as well on other OSs.  It should work everywhere with less install headache.

If you still want to pursue why your system failed here, I'd be curious to look at it a bit more in case it comes up again with some other fix on some other game (and I enjoy the challenge of a mystery), but it's not necessary at this point.

Thank you for helping me find the drive to look harder for a machine code fix.  Having better tools at hand lately also helped.

**Comment by TheRaccoonKing** [comment]
Sorry for my delayed response.  The last 3 days have been pure chaos for me.

At this point, what would help you?  If I can send you any helpful information, I'd be happy to assist.

**Comment by juanitogan** [comment]
New version of the patch out that should be easier to get running on more systems.  Check it out.  Closing as no longer relevant to the new tech in the patch.
